packages/editor/src/components/document-bar/index.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, isRTL } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	Button,
+	VisuallyHidden,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { BlockIcon } from '@wordpress/block-editor';
+import {
+	chevronLeftSmall,
+	chevronRightSmall,
+	page as pageIcon,
+	navigation as navigationIcon,
+	symbol,
+} from '@wordpress/icons';
+import { displayShortcut } from '@wordpress/keycodes';
+import { useEntityRecord } from '@wordpress/core-data';
+import { store as commandsStore } from '@wordpress/commands';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+const typeLabels = {
+	wp_pattern: __( 'Editing pattern:' ),
+	wp_navigation: __( 'Editing navigation menu:' ),
+	wp_template: __( 'Editing template:' ),
+	wp_template_part: __( 'Editing template part:' ),
+};
+
+const icons = {
+	wp_block: symbol,
+	wp_navigation: navigationIcon,
+};
+
+export default function DocumentBar() {
+	const { isEditingTemplate, templateId, postType, postId } = useSelect(
+		( select ) => {
+			const {
+				getRenderingMode,
+				getCurrentTemplateId,
+				getCurrentPostId,
+				getCurrentPostType,
+			} = select( editorStore );
+			const _templateId = getCurrentTemplateId();
+			return {
+				isEditingTemplate:
+					!! _templateId && getRenderingMode() === 'template-only',
+				templateId: _templateId,
+				postType: getCurrentPostType(),
+				postId: getCurrentPostId(),
+			};
+		},
+		[]
+	);
+	const { getEditorSettings } = useSelect( editorStore );
+	const { setRenderingMode } = useDispatch( editorStore );
+
+	return (
+		<BaseDocumentActions
+			postType={ isEditingTemplate ? 'wp_template' : postType }
+			postId={ isEditingTemplate ? templateId : postId }
+			onBack={
+				isEditingTemplate
+					? () =>
+							setRenderingMode(
+								getEditorSettings().defaultRenderingMode
+							)
+					: undefined
+			}
+		/>
+	);
+}
+
+function BaseDocumentActions( { postType, postId, onBack } ) {
+	const { open: openCommandCenter } = useDispatch( commandsStore );
+	const { editedRecord: document, isResolving } = useEntityRecord(
+		'postType',
+		postType,
+		postId
+	);
+	const { templateIcon, templateTitle } = useSelect( ( select ) => {
+		const { __experimentalGetTemplateInfo: getTemplateInfo } =
```

Comment:
I hate this selector, it's so bad in so many ways, but it's what we have today. Let's keep changes to that for later.

Reply:
- Well first it says `getTemplateInfo` and it works both for templates and template parts
 - I don't see why we need something like that specific to these post types but can't include all post types (to avoid the isTemplate conditionals in this component for instance)
 - It takes an object, why? It should take postType, postId. A selector taking an object like that is hard to memoize... The object is probably changing way too often.